morphir/src/org/finos/morphir/datamodel/Concept.scala

```diff
@@ -229,4 +234,115 @@ object Concept {
+    def defaultUnion(concept: Concept.Union) = default(concept.cases.head).map(Data.Union(_, concept))
+    def defaultRecord(concept: Concept.Record): DefaultEither = for {
+      dataFields <-
+        Defaults.collectAll(concept.fields, (label: Label, concept: Concept) => default(concept).map((label, _)))
+      res = Data.Record(dataFields, concept)
+    } yield res
+    def defaultStruct(concept: Concept.Struct): DefaultEither = for {
+      dataFields <- collectAll(concept.fields, (label: Label, concept: Concept) => default(concept).map((label, _)))
+      res = Data.Struct(dataFields)
+    } yield res
+
+    def defaultEnum(concept: Concept.Enum): DefaultEither = {
+      val firstCase = concept.cases.head
+      for {
+        args <- collectAll(firstCase.fields, (label: EnumLabel, concept: Concept) => default(concept).map((label, _)))
+        res = Data.Case(args, firstCase.label.value, concept)
+      } yield res
+    }
+    def defaultInt64: DefaultEither   = Right(Data.Int64(0))
+    def defaultInteger: DefaultEither = Right(Data.Integer(BigInt(0)))
+    def defaultDecimal: DefaultEither = Right(Data.Decimal(BigDecimal(0)))
+    def defaultInt16: DefaultEither   = Right(Data.Int16(0))
+    def defaultNothing: DefaultEither = Left(NoDefaultNothing())
+    def defaultMonth: DefaultEither   = Right(Data.Month(java.time.Month.JANUARY))
+    def defaultLocalDate: DefaultEither =
+      Right(Data.LocalDate(java.time.LocalDate.EPOCH))
+    def defaultFloat: DefaultEither = Right(Data.Float(0))
+    def defaultLocalTime: DefaultEither =
+      Right(Data.LocalTime(java.time.LocalTime.MIDNIGHT))
+    def defaultInt32: DefaultEither  = Right(Data.Int32(0))
+    def defaultByte: DefaultEither   = Right(Data.Byte(0))
+    def defaultUnit: DefaultEither   = Right(Data.Unit)
+    def defaultString: DefaultEither = Right(Data.String(""))
+    def defaultOrder: DefaultEither  = Right(Data.Order.apply(0))
+    def defaultDayOfWeek: DefaultEither =
+      Right(Data.DayOfWeek(java.time.DayOfWeek.MONDAY))
+    def defaultChar: DefaultEither    = Right(Data.Char('0'))
+    def defaultBoolean: DefaultEither = Right(Data.False)
+    def defaultOptional(concept: Concept.Optional): DefaultEither =
+      Right(Data.Optional.None(concept.elementType))
+    def defaultAny: DefaultEither = Right(Data.Unit)
+    def defaultResult(concept: Concept.Result): DefaultEither =
+      default(concept.okType).map(Data.Result.Ok(_, concept))
+    def defaultSet(concept: Concept.Set): DefaultEither = Right(Data.Set.empty(concept.elementType))
+    def defaultTuple(concept: Concept.Tuple): DefaultEither = for {
+      elems <- collectAll(concept.values, default)
+      res = Data.Tuple(elems)
+    } yield res
+  }
+
+  object Defaults {
+    type DefaultEither = Either[MDMDefaultError, Data]
+    sealed trait MDMDefaultError  extends Throwable
+    case class NoDefaultNothing() extends MDMDefaultError
+
+    def collectAll[I, O, E](
```

Comment:
Should this be publicly available? If it is an internal helper only make it `private` or package private via a `private[datamodel]` or `private[morphir]` access modifier.